internal/process/queue.go

```diff
@@ -96,8 +131,43 @@ func (q *Queue) worker(queue workqueue.RateLimitingInterface, process func(key s
 				}
 
 				queue.Forget(key)
+				log.Infof("item for %s has been processed, no retry, element forgotten", id)
+
 				return false
 			}()
 		}
 	}
 }
+
+func (q *Queue) logAndUpdateWorkerTimes(key string, name string, log logrus.FieldLogger) {
+	// log time
+	now := time.Now()
+	lastTime, ok := q.workerExecutionTimes[name]
+	if ok {
+		log.Infof("execution - worker %s last execution time %s, executed after %s", name, lastTime, now.Sub(lastTime))
+	}
+	q.workerExecutionTimes[name] = now
+
+	log.Infof("processing item %s, queue length is %d", key, q.queue.Len())
+}
+
+func (q *Queue) HealthCheck() {
```

Comment:
This `health check` checks interval between `now` and the last time workers processed the queue element. 

What is the intended purpose? What should this health check detect?
Would it detect long processing?
Would it detect worker not being called by called?

Reply:
This is to detect how many workers are stuck, whatever a reason may be.

Would it detect long processing? - yes
Would it detect worker not being called by wait.Until? - yes
What would happen if queue is empty for some time? - that is a good point, which made me change warn level to info

Please, treat as information only

Reply:
My point is that it detects long time between subsequent `logAndUpdateWorkerTimes` calls.
The reason could be:
- empty queue
- worker not being selected on blocking `Get` (other worker selected)
- long processing 
- not being called by `wait.Until`

Reply:
I really need this line to quickly find what was going on. In all cases mentioned by you if Info is logged it means that we either could adjust workers number or we have some unexpected behaviour.